sensor/common/scan/scan.go

```diff
@@ -109,6 +112,12 @@ func NewLocalScan(registryStore registryStore, mirrorStore registrymirror.Store)
 		getPullSecretRegistries:   registryStore.GetPullSecretRegistries,
 		getGlobalRegistry:         registryStore.GetGlobalRegistry,
 	}
+	// allow certain number of parallel delegated scans
+	if !env.DelegatedScanningDisabled.BooleanSetting() {
+		ls.adHocScanSemaphore = semaphore.NewWeighted(int64(env.MaxParallelAdHocScanInternal.IntegerSetting()))
+		ls.scanSemaphore = semaphore.NewWeighted(int64(max(imageScanLowerBound, env.MaxParallelImageScanInternal.IntegerSetting()-env.MaxParallelAdHocScanInternal.IntegerSetting())))
+	}
```

Comment:
This `env.DelegatedScanningDisabled.BooleanSetting()` check should not be necessary.  Let me know if you see otherwise.